ci/docsgen/schema.py

```diff
@@ -0,0 +1,144 @@
+    title: The title of the document.
+    metadata: Additional metadata to include in the document.
+
+  Returns:
+    A flavoured markdown document representing the schema.
+  """
+
+  flat_schema = flatten_dict(schema)
+  flat_sample = flatten_dict(sample)
+
+  table_of_contents = '<h2 id=table-of-contents>Table of Contents</h2>'
+  table_of_contents += '\n\n<details><summary>Click to Expand</summary>\n'
+
+  document = ''
+
+  for key, entry in sorted(raw_schema.items()):
+    entry_fmt = format_markdown_entry(key, entry)
+
+    # Add friendlier anchor links for each entry
+    # E.g. 'ACPI -> Add[] -> Comment' -> '#acpi-add-comment'
+    level, name = re_search(r'^([#]*)\s+(.*?)$', entry_fmt, None, True).groups()
+    slug = str(name) \
+      .replace(' -> ', '-') \
+      .replace('[]', '') \
+      .lower()
+    header = f'<h{len(level)} id={slug}>{name}</h{len(level)}>'
+    entry_fmt = "\n".join([header, *entry_fmt.split('\n')[1:]])
+
+    # Add entry to the table of contents
+    list_indent = ' ' * 2 * (len(level) - 2)
+    table_of_contents += f'\n{list_indent}- [{name}](#{slug})'
+
+    # Extract defaults from Sample.plist or Schema.plist
+    default = nested_get(flat_sample, (key,), default=None)
+    if default is not None and '[]' in name:
+      default = nested_get(flat_schema, (key,), default=None)
+
+    # Add a default value to the entry
+    if default is not None:
+      if isinstance(default, (str, bytes)) and not len(default):
+        default = 'Empty'
+      elif isinstance(default, str) and len(default):
+        default = f"`{default}`"
+      elif isinstance(default, bytes) and len(default):
+        default = f"`0x{decode_data(default, enc='hex')}`"
+      elif isinstance(default, bool):
+        default = f"`{str(default).lower()}`"
+      elif isinstance(default, (int, float)):
+        default = f"`{default}`"
+      elif isinstance(default, dict):
+        default = '`plist dict`' if default else 'Empty'
+      elif isinstance(default, list):
+        default = '`plist array`' if default else 'Empty'
+
+      # Insert default value before the failsafe value.
+      entry_fmt = entry_fmt \
+        .replace(ins := '**Failsafe**:', f'**Default**: {default}\n\n{ins}')
+
+    document += f"\n\n{entry_fmt.rstrip()}"
+
+  header = f'<h1 id=schema>{title}</h1>'
+  # Add metadata to the header
+  if metadata is not None:
+    version = metadata.get('version', 'Unknown')
+    timestamp = datetime.now(tz=timezone.utc)
+    revision = metadata.get('revision', 'Unknown')
+
+    header = f"<h1 id=schema>{title} - v{version}</h1>"
+    header += f"\n\n#### Last Updated: `{timestamp}`\n"
+    header += f"\n#### Revision: `{revision}`"
+
+  table_of_contents += '\n\n</details>'
+  document = f"{header}\n\n{table_of_contents}{document}"
+
+  return document
+
+
+if __name__ == '__main__':
+  entry = next(iter(resolve_specifiers({
+    'OpenCorePkg': {
+      'OpenCore': {
+        'specifier': 'latest',
```

Comment:
Schema docs will need to be updated on a regular cadence following each OpenCore release.

Pre-release builds of OpenCore will need to be smoke-tested against any schema validation in case new schema entries are added that must be specially handled.